### PR TITLE
Refactor core/Command/Background

### PR DESCRIPTION
**Pull request by shdehnavi**
## Summary
The required adjustments have been made to the classes in `/core/Command/Background ` namespace.

The improvements:

- Adding return types
- Adding throws types
- Updating doc blocks
- 
## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

File: core/Command/Background/Job.php [hunk 1]
```diff
@@ -104,7 +107,10 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 		return 0;
 	}
 
-	protected function printJobInfo(int $jobId, IJob $job, OutputInterface$output): void {
+	/**
+	 * @throws \ReflectionException
```
Comment (ChristophWurst) [inline core/Command/Background/Job.php:111]: I wouldn't document because this is not an exception to expect/catch when calling the method

Comment (shdehnavi) [reply core/Command/Background/Job.php]: It's addressed in the next commit

**Review by skjnldsv — APPROVED** [review]

File: core/Command/Background/Job.php [hunk 2]
```diff
@@ -60,6 +60,9 @@ protected function configure(): void {
 		;
 	}
 
+	/**
+	 * @throws \ReflectionException
```
Comment (ChristophWurst) [inline core/Command/Background/Job.php:64]: same here. I would not document this.

Comment (shdehnavi) [reply core/Command/Background/Job.php]: It's addressed in the next commit

**Review by ChristophWurst — APPROVED** [review]
Squash your commits into one please

:+1: thanks

**Pull request merged**